File: frontend/lib/features/feed/domain/entities/post.dart
class Post {
  const Post({
    required this.id,
    required this.title,
    required this.message,
    required this.creatorId,
    required this.creatorName,
    required this.creatorUsername,
    required this.creatorImageUrl,
    required this.selectedFile,
    required this.likes,
    required this.createdAt,
  });

  final String id;
  final String title;
  final String message;
  final String creatorId;
  final String creatorName;
  final String creatorUsername;
  final String creatorImageUrl;
  final String? selectedFile;
  final List<String> likes;
  final DateTime createdAt;

  bool likedBy(String userId) => likes.contains(userId);
}

File: backend/api/models/post.py
from datetime import datetime, timezone
from typing import List, Optional
from beanie import Document
from pydantic import Field
import pymongo


class Post(Document):
    title: str
    message: str
    creator: str
    selectedFile: Optional[str] = Field(default="")
    likes: Optional[List[str]] = Field(default=[])
    createdAt: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))

    class Settings:
        name = "posts"
        indexes = [
            [("title", pymongo.TEXT), ("message", pymongo.TEXT)],
            [("creator", 1), ("createdAt", -1)],
        ]